test/specs/rules/disallow-quoted-keys-in-objects.js

```diff
@@ -67,4 +67,12 @@ describe('rules/disallow-quoted-keys-in-objects', function() {
 
         assert(checker.checkString('var x = { "null": 1, undefined: "foo" }').isEmpty());
     });
+
+    it('should report non-reserved quoted keys when "allButReserved" mode is used. #1669', function() {
```

Comment:
Deprecating `allButReserved `? Yeah, that would be cool.

> Should we do it now or... in other PR?

If that is not too much trouble